### PR TITLE
Kotlin Sync Feedback Document changes

### PR DESCRIPTION
**Pull request by edualonso**
Incorporated changes from https://docs.google.com/document/d/1M2QoA-auyOaWZ88PeL1_6QylcEkqt3Aqv3ppOvuuGtI/edit#heading=h.watv1kfii35

### CODE REVIEW

File: source/sdk/kotlin/sync/overview.txt [hunk 1]
```diff
@@ -21,11 +21,14 @@ device is online, Sync asynchronously synchronizes data in a
...
-the :ref:`Sync <sync-modes>` you select in your backend App configuration.
+the :ref:`Sync Mode <sync-modes>` you select in your backend App configuration.
 The Sync Mode options are:
 
-- Partition-Based Sync
-- Flexible Sync
+- Partition-Based Sync: documents in a synced cluster form a "partition" by
+  having the same value for a field designated as "partition key". All documents
+  in a partition have the same read/write permissions for a given user.
+- Flexible Sync: lets you define a query in the client and sync only the objects
```
Comment (cbush) [inline source/sdk/kotlin/sync/overview.txt:30]: I would put flex sync first and even say "Flexible Sync (recommended)".

File: source/sdk/kotlin/install/kotlin-multiplatform.txt [hunk 2]
```diff
@@ -15,6 +15,12 @@ Additionally, Kotlin Multiplatform projects require the following:
   Studio. Follow the instructions in the `KMM documentation
   <https://kotlinlang.org/docs/mobile/create-first-app.html>`__.
 
+You can find a more detailed explanation of the steps you should follow
+to set up your KMM environment `here
+<https://kotlinlang.org/docs/multiplatform-mobile-setup.html>`__,
+and especially the section on
+`how to check you're all set up <https://kotlinlang.org/docs/multiplatform-mobile-setup.html#check-your-environment>`__.
```
Comment (cbush) [inline source/sdk/kotlin/install/kotlin-multiplatform.txt:22]: I would rewrite this as:

```
For more details on setting up your KMM environment, see `the official Kotlin
Multiplatform documentation
<https://kotlinlang.org/docs/multiplatform-mobile-setup.html>`__. To verify your
environment setup, follow Kotlin's `guide to checking your
environment
<https://kotlinlang.org/docs/multiplatform-mobile-setup.html#check-your-environment>`__.
```

General tips:
- We would avoid setting the link text to "here". Instead, be transparent about where you're sending the reader.
- Each idea should get its own sentence, more or less.

File: source/examples/generated/kotlin/QuickStartTest.snippet.quick-start-open-a-local-realm.kt [hunk 3]
```diff
@@ -1,3 +1,3 @@
-val config = RealmConfiguration.Builder(schema = setOf(Item::class))
+val config = RealmConfiguration.create(schema = setOf(Item::class))
```
Comment (cbush) [inline source/examples/generated/kotlin/QuickStartTest.snippet.quick-start-open-a-local-realm.kt:1]: Why this change? Anyway, it must be updated here: [examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/QuickStartTest.kt](https://github.com/mongodb/docs-realm/blob/master/examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/QuickStartTest.kt#L121)

**Review by cbush — COMMENTED** [review]
Thanks!

File: source/examples/generated/kotlin/QuickStartTest.snippet.quick-start-open-a-local-realm.kt (hunk 3 above)
Comment (edualonso) [reply source/examples/generated/kotlin/QuickStartTest.snippet.quick-start-open-a-local-realm.kt:1]: It's a fast-tracked convenience way to create a builder if no other config parameters are used. Which makes sense for a quick start section.

Comment (edualonso) [reply source/examples/generated/kotlin/QuickStartTest.snippet.quick-start-open-a-local-realm.kt:1]: Which means it cannot be used in https://github.com/mongodb/docs-realm/blob/master/examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/QuickStartTest.kt#L121 as it's using `directory` and `name` modifiers.

**Comment by edualonso** [comment]
The PR should now be good to go, waiting for final approval @cbush

File: source/examples/generated/kotlin/QuickStartTest.snippet.quick-start-open-a-local-realm.kt (hunk 3 above)
Comment (cbush) [reply source/examples/generated/kotlin/QuickStartTest.snippet.quick-start-open-a-local-realm.kt:1]: The problem is that the next time someone runs bluehawk on this, this change will get overridden. Anything in a generated/ directory can't be modified manually.
